inc/switchtec/mfg.h

```diff
@@ -92,6 +92,33 @@ struct switchtec_security_cfg_state {
 	uint8_t public_key[SWITCHTEC_KMSK_NUM][SWITCHTEC_KMSK_LEN];
 };
 
+struct switchtec_security_cfg_state_gen5 {
```

Comment:
This is a public struct and is almost identical to the one above it.

We really shouldn't be creating generation specific interfaces unless there is a really good reason to.

Seems like the library should be able to handle all the differences here instead of creating massive duplication all over the place....